setup.py

```diff
@@ -30,20 +30,21 @@ def read(*filenames, **kwargs):
 )
 testing = "test" in sys.argv or "pytest" in sys.argv
 
+numpy = "numpy>=1.8,<2"
 docs_require = [
-    "jupyter_client",
-    "sphinx",
-    "nbsphinx",
-    "nengo_sphinx_theme",
+    "jupyter_client>=5.3.4,<7",
+    "sphinx>=2.2.1,<4",
+    "nbsphinx>=0.5.0,<0.8",
+    "nengo_sphinx_theme>=1.2.0,<2",
 ]
-optional_requires = ["scipy"]
+optional_requires = ["scipy>=1.4.1,<2"]
 tests_require = [
-    "jupyter",
-    "matplotlib>=2.0",
-    "nbformat",
-    "pytest>=4.0,<5",
-    "pytest-plt",
-    "pytest-rng",
+    "jupyter>=1.0.0,<2",
+    "matplotlib>=2.0,<3",
```

Comment:
- [ ] Just noticed that there is already a version 3 of matplotlib, I should check whether that works fine.